src/test/java/org/example/integration/ApplicationIntegrationTest.java

```diff
@@ -0,0 +1,192 @@
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+
+@ExtendWith(DropwizardExtensionsSupport.class)
+public class ApplicationIntegrationTest {
+    static final DropwizardAppExtension<TestConfiguration> APP = new DropwizardAppExtension<>(
+            TestApplication.class, null,
+            new ResourceConfigurationSourceProvider()
+    );
+
+    @Test
+    public void getSingleApplication_shouldReturn200_whenGettingExistingApplication() {
+        Client client = APP.client();
+
+        LoginRequest loginRequest = new LoginRequest(
+                "admin@kainos.com",
+                "wlSNgEn5dCBM59jnbeH+txKWn36Vt6QScELcAa5ZBNduqSY16JAl2hqeGsZrmpG0kdb9+ILMoCJVB3er8ZoCJI9o26IM83UfnJtTT3p7cRgOUxsU0iMHgkI9KdQpDim6"
+
+        );
+        String token = "Bearer " + client
+                .target(System.getenv("API_URL")+"auth/login")
+                .request()
+                .post(Entity.json(loginRequest))
+                .readEntity(String.class);
+
+
+        Response response = client
+                .target(System.getenv("API_URL")+"apply/1")
+                .request()
+                .header(HttpHeaders.AUTHORIZATION, token)
+                .post(Entity.json("user@kainos.com"));
+
+        Assertions.assertEquals(200, response.getStatus());
+
+    }
+
+    @Test
+    public void getSingleApplication_shouldReturn401_whenUnauthenticated() {
+        Client client = APP.client();
+
+        Response response = client
+                .target(System.getenv("API_URL")+"apply/1")
+                .request()
+                .post(Entity.json("user@kainos.com"));
+
+        Assertions.assertEquals(401, response.getStatus());
+
+    }
+
+    @Test
+    public void getSingleApplication_shouldReturn404_whenApplicationDoesNotExist() {
+        Client client = APP.client();
+
+        LoginRequest loginRequest = new LoginRequest(
+                "admin@kainos.com",
+                "wlSNgEn5dCBM59jnbeH+txKWn36Vt6QScELcAa5ZBNduqSY16JAl2hqeGsZrmpG0kdb9+ILMoCJVB3er8ZoCJI9o26IM83UfnJtTT3p7cRgOUxsU0iMHgkI9KdQpDim6"
+
+        );
+
+        String token = "Bearer " + client
+                .target(System.getenv("API_URL")+"auth/login")
+                .request()
+                .post(Entity.json(loginRequest))
+                .readEntity(String.class);
+
+
+        Response response = client
+                .target(System.getenv("API_URL")+"apply/-1")
+                .request()
+                .header(HttpHeaders.AUTHORIZATION, token)
+                .post(Entity.json("user@kainos.com"));
+
+        Assertions.assertEquals(404, response.getStatus());
+
+    }
+
+    @Test
+    public void createApplication_shouldReturn201_whenCreatingSuccessfulApplication() {
+        ApplicationRequest applicationRequest = new ApplicationRequest(
+                "admin@kainos.com",
+                1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
```

Comment:
Look at previous comments, remove URL as you are exposing it to the world. Everyone knows, the polar bears and pirates are looking at the S3 bucket now. They are uploading their own CVs and you don't want to work with a polar bear. You just want to save them.